cpp/src/parquet/page_index.cc

```diff
@@ -309,6 +313,33 @@ class RowGroupPageIndexReaderImpl : public RowGroupPageIndexReader {
     }
   }
 
+  /// Get decryptor if the file is encrypted
+  std::shared_ptr<Decryptor> GetColumnMetaDecryptor(
+      const ColumnCryptoMetaData* crypto_metadata, int column_ordinal,
+      int8_t module_type) const {
+    std::shared_ptr<Decryptor> decryptor;
+    if (crypto_metadata != nullptr) {
```

Comment:
consider changing this to a guard to remove one level of nesting in the main function.